src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -88,6 +100,46 @@ module.exports = function<T, P, I, TI, C, CX>(
     updateClassInstance,
   } = ReactFiberClassComponent(scheduleUpdate, getPriorityContext);
 
+  function isValidNode(node) {
```

Comment:
I don't like that this is a duplicate runtime type check with the branch check that is done in ReactChildFiber.

IMO we should move this validation into the ReactChildFiber since you can just throw if you fall out into the last branch.